src/shared/style/utils.ts

```
@@ -1,6 +1,53 @@
+  props: T,
+  variantKeys?: K[],
+  removeVariantProps?: R,
+): readonly [Omit<T, K> | T, Pick<T, K>] {
+  if (!variantKeys) {
+    return [props, {} as Pick<T, K>];
+  }
+
+  const picked = variantKeys.reduce((acc, key) => {
+    // Only include the key in `picked` if it exists in `props`
+    if (key in props) {
+      return { ...acc, [key]: props[key] };
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)